src/celestia/hud.h

```diff
@@ -0,0 +1,193 @@
+
+#include <Eigen/Core>
+
+#include <celengine/astro.h>
+#include <celengine/dateformatter.h>
+#include <celengine/selection.h>
+#include <celestia/textinput.h>
+#include <celestia/textprintposition.h>
+#include <celestia/windowmetrics.h>
+#include <celttf/truetypefont.h>
+#include <celutil/color.h>
+#include <celutil/flag.h>
+
+class MovieCapture;
+class Overlay;
+class OverlayImage;
+class Simulation;
+
+namespace celestia
+{
+
+class TimeInfo;
+class ViewManager;
+
+namespace engine
+{
+class DateFormatter;
+}
+
+enum class MeasurementSystem
+{
+    Metric      = 0,
+    Imperial    = 1,
+};
+
+enum class TemperatureScale
+{
+    Kelvin      = 0,
+    Celsius     = 1,
+    Fahrenheit  = 2,
+};
+
+class HudFonts
+{
+public:
+    HudFonts() = default;
+
+    void setFont(const std::shared_ptr<TextureFont>&);
+    void setTitleFont(const std::shared_ptr<TextureFont>&);
+
+    inline const std::shared_ptr<TextureFont>& font() const noexcept { return m_font; }
+    inline const std::shared_ptr<TextureFont>& titleFont() const noexcept { return m_titleFont; }
+    inline int fontHeight() const noexcept { return m_fontHeight; }
+    inline int titleFontHeight() const noexcept { return m_titleFontHeight; }
+    inline int emWidth() const noexcept { return m_emWidth; }
+    inline int titleEmWidth() const noexcept { return m_titleEmWidth; }
+
+private:
+    std::shared_ptr<TextureFont> m_font;
+    std::shared_ptr<TextureFont> m_titleFont;
+    int m_fontHeight{ 0 };
+    int m_titleFontHeight{ 0 };
+    int m_emWidth{ 0 };
+    int m_titleEmWidth{ 0 };
+};
+
+enum class HudElements : int
+{
+    ShowTime       = 0x01,
+    ShowVelocity   = 0x02,
+    ShowSelection  = 0x04,
+    ShowFrame      = 0x08,
+    Default        = 0x0f,
+};
+
+ENUM_CLASS_BITWISE_OPS(HudElements);
+
+struct HudSettings
+{
+    Color textColor{ 1.0f, 1.0f, 1.0f };
+
+    MeasurementSystem measurementSystem{ MeasurementSystem::Metric };
+    TemperatureScale temperatureScale{ TemperatureScale::Kelvin };
+
+    HudElements overlayElements{ HudElements::Default };
+    bool showFPSCounter{ false };
+    bool showOverlayImage{ true };
+    bool showMessage{ true };
+};
+
+class Hud
+{
+public:
+
+
+    enum class TextEnterMode : unsigned int
+    {
+        Normal       = 0x00,
+        AutoComplete = 0x01,
+        PassToScript = 0x02,
+    };
+
+    Hud() = default;
+    ~Hud();
+
+    int detail() const;
+    void detail(int);
+    astro::Date::Format dateFormat() const;
+    void dateFormat(astro::Date::Format);
+    TextInput& textInput();
+    TextEnterMode textEnterMode() const;
+    void textEnterMode(TextEnterMode mode);
+
+    void setOverlay(std::unique_ptr<Overlay>&&);
+    void setWindowSize(int, int);
+    void setTextAlignment(LayoutDirection);
+
+    int getTextWidth(std::string_view) const;
+    const std::shared_ptr<TextureFont>& font() const;
+    const std::shared_ptr<TextureFont>& titleFont() const;
+    bool trySetFont(const std::shared_ptr<TextureFont>&);
```

Comment:
why the hud should care about value of `f`? i think it's a job of the caller, so i propose jusr `setFont` & `setTitleFont`